src/main.cpp

```diff
@@ -1032,8 +1039,38 @@ bool ContextualCheckTransaction(
         // after Canopy activation.
     }
 
-    auto consensusBranchId = CurrentEpochBranchId(nHeight, chainparams.GetConsensus());
-    auto prevConsensusBranchId = PrevEpochBranchId(consensusBranchId, chainparams.GetConsensus());
+    // Rules that apply to the future epoch
+    if (futureActive) {
+        // version group is the most recent available version group ID
```

Comment:
This comment doesn't make sense to me. `ZFUTURE_VERSION_GROUP_ID` isn't an "available" version group ID.